features/datachoices/src/main/java/org/opennms/features/datachoices/internal/UsageStatisticsReporter.java

```diff
@@ -80,6 +83,10 @@
 
 @Service
 public class UsageStatisticsReporter implements StateChangeHandler {
+    
+    @Reference
```

Comment:
@j-white I thought that this annotation would take care of wiring in the KarafExtender, but it's ending up null as evidenced by the NPE thrown every time `generateReport()` runs. Does the KarafExtender class perhaps need an `@Service` annotation to get it into the registry?

Reply:
The `@Service` and `@Reference` annotations should only be used for the shell commands AFAIK.

Also, the `KarafExtender` bean isn't being exported in the OSGi registry currently, so there is no way of referencing it. You would need to update blueprint to expose it, and then reference it, or re-declare it altogether in the datachoices features.